integrationtests/tests/fixtures/mod.rs

```diff
@@ -1129,44 +1138,72 @@ impl FederationTest {
     // will never run an epoch concurrently with itself.
     #[allow(clippy::await_holding_refcell_ref)]
     async fn consensus_epoch(
-        server: Rc<RefCell<ServerTest>>,
+        server: Arc<Mutex<ServerTest>>,
         delay: Duration,
     ) -> anyhow::Result<()> {
         tokio::time::sleep(delay).await;
-        let mut s = server.borrow_mut();
-        let consensus = s.fedimint.consensus.clone();
+        let mut server = server.lock().await;
+        let consensus = server.fedimint.consensus.clone();
 
-        let overrider = s.override_proposal.clone();
+        let overrider = server.override_proposal.clone();
         let proposal = async { overrider.unwrap_or(consensus.get_consensus_proposal().await) };
-        s.dropped_peers
+        server
+            .dropped_peers
             .append(&mut consensus.get_consensus_proposal().await.drop_peers);
 
-        s.last_consensus = s.fedimint.run_consensus_epoch(proposal, &mut rng()).await?;
+        server.last_consensus = server
+            .fedimint
+            .run_consensus_epoch(proposal, &mut rng())
+            .await?;
 
-        for outcome in s.last_consensus.clone() {
-            s.fedimint.process_outcome(outcome).await.expect("failed");
+        for outcome in server.last_consensus.clone() {
+            server
+                .fedimint
+                .process_outcome(outcome)
+                .await
+                .expect("failed");
         }
 
         Ok(())
     }
 
-    fn update_last_consensus(&self) {
-        let new_consensus = self
-            .servers
-            .iter()
-            .flat_map(|s| s.borrow().last_consensus.clone())
-            .max_by_key(|c| c.epoch)
+    async fn update_last_consensus(&self) {
+        let new_consensus = futures::stream::iter(self.servers.iter())
+            .then(|s| s.lock())
```

Comment:
Yes. It will become a part of some big thing that implements `Stream` that will have `.fold` ran on it, which consumes it, which calls drops on everything inside it.

Reply:
you can't answer an option A *or* option B question with "yes" :'D 

so the lock is dropped at the end of `flat_map` after the value is cloned because the look is not needed afterwards, right?